### PR TITLE
DA-314: add readme for periph_board

### PR DESCRIPTION
**Pull request by mcvella**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+## Build
+
+On the single board computer that will be running viam-server and the periph_board module, run these commands to install prerequisites and build:
+
+``` bash
+sudo apt update && sudo apt upgrade -y
+sudo apt-get install git
+wget https://go.dev/dl/go1.20.5.linux-arm64.tar.gz
+sudo  rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.20.5.linux-arm64.tar.gz
+echo 'PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && source ~/.bashrc
```
Comment (biotinker) [inline README.md:14]: Not sure we need the Go install instructions included, especially since they're specific to a version of go, and to linux amd64.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+Enable i2c and spi by editing `/boot/orangepiEnv.txt`, adding the line:
+
+``` bash
+overlays=i2c3 spi-spidev
+param_spidev_spi_bus=1
+param_spidev_spi_cs=1
+```
+
+Now install the Orange Pi gpio tool:
```
Comment (biotinker) [inline README.md]: The above overlays are actually needed for Viam to function properly, but the below WiringOP stuff should be fully unrelated to Viam and is just a coincident debug tool; not sure it needs to be part of the setup.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# periph_board modular board component
+
+*periph_board* is a Viam modular board component that an periph.io based GPIO interface for single board computers.
```
Comment (penguinland) [inline README.md]: Grammar: "an periph.io"

Grammar: it's a board component that...for computers. Needs another verb in the "..."

File: README.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+## Build
+
+On the single board computer that will be running viam-server and the periph_board module, run these commands to install prerequisites and build:
```
Comment (penguinland) [inline README.md:7]: Rather than trying to build and install everything on some tiny, underpowered board, I'd grab [our `canon` utility](https://github.com/viamrobotics/canon) (by running `go install github.com/viamrobotics/canon@latest`, rather than cloning the repo), and then in this directory just run `canon -arch arm64`, then `go build` inside the shell it opens (if you're not running it on an arm64 machine, you'll need to enable architecture emulation, but the github page linked above has instructions for this). This sets up and configures everything automatically, and you can do it on a more powerful computer like a laptop and then copy the binary onto the SBC.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+``` bash
+sudo apt update && sudo apt upgrade -y
+sudo apt-get install git
+wget https://go.dev/dl/go1.20.5.linux-arm64.tar.gz
+sudo  rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.20.5.linux-arm64.tar.gz
```
Comment (penguinland) [inline README.md:13]: This permanently changes the version of Go running on a user's machine, in a way that will likely go unnoticed by `apt` (so, very hard to undo later). If the user is relying on a different version of Go for something else, they're gonna have a really bad time. I have a strong preference for using any other option besides this line (for example, going the `canon` route, though other options might exist, too).

File: README.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    }
+  ]
+}
+```
+
+## viam-server board setup examples
+
+### Orange Pi Zero 2
+
+The [Orange Pi Zero 2](http://www.orangepi.org/html/hardWare/computerAndMicrocontrollers/details/Orange-Pi-Zero-2.html) is an open-source SBC with dual-band wireless WiFi and Bluetooth 5.0 .It is highly compact with a dimension of 53X60mm. It can run Android10, Ubuntu, Debian.
```
Comment (penguinland) [inline README.md]: Swap the period and space: "Bluetooth 5.0 .It"

Grammar: put an "and" before "Debian".

File: README.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+Start by downloading the latest Linux distribution from OrangePi.
+We tested and had success with the [Debian bullseye distribution](https://drive.google.com/drive/folders/1Xk7b1jOMg-rftowFLExynLg0CyuQ7kCM).
+
+Flash this image to an SD card with [Balena Etcher](https://etcher.balena.io/)
+
+Then insert the flashed SD card to the Orange Pi, plug the Pi into a monitor with an mini-HDMI cable, and plug in the Pi to with a 5v power supply.
```
Comment (penguinland) [inline README.md]: Typo: "the pi to with"

File: README.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+Log in with the credentials root/orangepi.
+
+Secure the Pi:
+
+- Change the root password by issuing the `passwd` command.
+- Change the password for the `orangepi` user account by issuing the commmand `passwd orangepi`
```
Comment (penguinland) [inline README.md:100]: Securing the device is good advice, but it seems outside the scope of a readme for this component. Could we remove it?

I worry about the people who see the length of this readme and get so intimidated they give up, without even looking through it.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+Update the software:
+
+``` bash
+apt update
+apt upgrade
```
Comment (penguinland) [inline README.md:106]: Same suggestion here: this is good advice but has nothing to do with setting up the periph_board component. I'd remove it as off-topic.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+The [Orange Pi Zero 2](http://www.orangepi.org/html/hardWare/computerAndMicrocontrollers/details/Orange-Pi-Zero-2.html) is an open-source SBC with dual-band wireless WiFi and Bluetooth 5.0 .It is highly compact with a dimension of 53X60mm. It can run Android10, Ubuntu, Debian.
+
+Start by downloading the latest Linux distribution from OrangePi.
+We tested and had success with the [Debian bullseye distribution](https://drive.google.com/drive/folders/1Xk7b1jOMg-rftowFLExynLg0CyuQ7kCM).
+
+Flash this image to an SD card with [Balena Etcher](https://etcher.balena.io/)
+
+Then insert the flashed SD card to the Orange Pi, plug the Pi into a monitor with an mini-HDMI cable, and plug in the Pi to with a 5v power supply.
+Log in with the credentials root/orangepi.
```
Comment (penguinland) [inline README.md:95]: Mild preference to encourage people to log in with the username `orangepi` (and password `orangepi`), rather than running as `root` all the time. You can use `sudo` when you need `root`like power, and I've wiped my system with a typo running as root before.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+``` bash
+apt update
+apt upgrade
+```
+
+Enable wifi:
+
+```bash
+nmcli dev wifi
+nmcli dev wifi connect <wifi_name> password <wifi_passwd>
```
Comment (penguinland) [inline README.md:113]: Same suggestion here about removing as off-topic: it's good advice, but I think it's off-topic. People can plug in an ethernet cable and skip this step, or else they'll want to do this step even without using Viam. I think we should emphasize how easy Viam is to use, by showing how little (Viam-specific) setup is required.

File: README.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+Enable the host being announced on the network by running `orangepi-config` and going to System->Avahi and *Enable system announce in the network*.
```
Comment (penguinland) [inline README.md:116]: Why is this needed at all? The `viam-server` process calls out to app.viam.com. If you don't need to SSH in to the SBC, I don't think we need to announce the host. (It's possible the correct response is "Alan, this really is needed, because you've never had a good grasp of how the networking part works. Here's what's really going on...")

File: README.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+While still in `orangepi-config`, change the hostname to your preference by going to Personal->Hostname.
```
Comment (penguinland) [inline README.md:118]: Again, good advice but off-topic.

I imagine you've written this for an audience of people who have just bought their first SBC, their first Linux system, and they don't know anything about either one yet. What would you say to having 2 files: this README.md with Viam-specific instructions, which links to a second, optional one about how to set up a basic system? That second one is where you tell folks how to change the hostname, update the software, set the password, get on the wifi, etc. That way, someone interested in just the Viam parts of this repo don't see the size of the readme and say "wow, Viam is super complicated," but a complete newcomer doesn't see how short it is and say "wow, Viam is only for experts." Viam is not only for experts, and also not complicated to set up.

File: README.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+
+Enable i2c and spi by editing `/boot/orangepiEnv.txt`, adding the line:
```
Comment (penguinland) [inline README.md]:
```suggestion
If you want to use the I2C or SPI buses, enable i2c and spi by editing `/boot/orangepiEnv.txt`, adding the lines:
```

File: README.md [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+
+``` bash
+git clone https://github.com/orangepi-xunlong/wiringOP
+cd wiringOP
+sudo ./build clean
+sudo ./build
+gpio readall
+```
+
+At this point, you can unplug the HDMI cable and restart.
```
Comment (penguinland) [inline README.md:138]: Why do you need to restart? If it's to pick up the devicetree overlays for SPI and I2C, move this up 2 paragraphs.

I'd remove the part about the HDMI cable entirely; it should be obvious to people when they need it.

File: README.md [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+You should now be able to connect to the Pi headless via `ssh <user>@<hostname>.local` and [install viam-server](https://docs.viam.com/installation/#install-viam-server)
```
Comment (penguinland) [inline README.md:139]: You should have been able to do that long before, without the I2C, SPI, and GPIO stuff. I'd put this up several paragraphs.

File: README.md [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+
+### Rock Pi S
+
+The [Rock Pi S](https://wiki.radxa.com/RockpiS) is a Rockchip RK3308 based SBC(Single Board Computer) by Radxa.
+It equips a 64bits quad core processor, USB, ethernet, wireless connectivity and voice detection engine at the size of 1.7 inch, makes it perfect for IoT and voice applications.
```
Comment (penguinland) [inline README.md]: Possible typo: 64bits

...this is copied and pasted from the URL on the previous line. I can't tell if that's okay, or plagiarism.

File: README.md [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+sudo ./build
+gpio readall
+```
+
+At this point, you can unplug the HDMI cable and restart.
+You should now be able to connect to the Pi headless via `ssh <user>@<hostname>.local` and [install viam-server](https://docs.viam.com/installation/#install-viam-server)
+
+### Rock Pi S
+
+The [Rock Pi S](https://wiki.radxa.com/RockpiS) is a Rockchip RK3308 based SBC(Single Board Computer) by Radxa.
```
Comment (penguinland) [inline README.md]: Typo: need a space between SBC and the open-parenthesis.

File: README.md [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+cd wiringOP
+sudo ./build clean
+sudo ./build
+gpio readall
+```
+
+At this point, you can unplug the HDMI cable and restart.
+You should now be able to connect to the Pi headless via `ssh <user>@<hostname>.local` and [install viam-server](https://docs.viam.com/installation/#install-viam-server)
+
+### Rock Pi S
```
Comment (penguinland) [inline README.md:141]: Ohh, I think I get it now! This PR is a combination readme for the module, and setup guide for 2 different boards! What would you feel about making it 3 different files instead? The readme could link to the other two as example setups, but then it's clear which parts are about this component and which are specific to boards that the user might not even own.

File: README.md [hunk 20]
```diff
@@ -0,0 +1,179 @@
...
+
+The [Rock Pi S](https://wiki.radxa.com/RockpiS) is a Rockchip RK3308 based SBC(Single Board Computer) by Radxa.
+It equips a 64bits quad core processor, USB, ethernet, wireless connectivity and voice detection engine at the size of 1.7 inch, makes it perfect for IoT and voice applications.
+
+Start by downloading the latest Armbian distribution for the RockPi S.
+We tested and had success with the [Bookwork distribution](https://www.armbian.com/rockpi-s/).
+
+Flash this image to an SD card with [Balena Etcher](https://etcher.balena.io/)
+
+Then insert the flashed SD card to the Rock Pi, plug the Pi into a router with an ethernet cable, and plug in the Pi to with a 5v power supply.
```
Comment (penguinland) [inline README.md]: Same typo here: "the Pi to with"

**Review by penguinland — COMMENTED** [review]
Many of my comments are the same misunderstanding over and over. I think it would be clearer if this was split into 3 files: a readme about this modular component, and 2 startup guides for the Orange Pi and Rock Pi.

File: README.md (hunk 5 above)
Comment (mcvella) [reply README.md:13]: I was assuming they were doing in on a cleanly installed linux on an SBC.  The canon recommendation makes sense, but I will leave as-is for now as I want to get this README merged and I have not yet tested the way you recommend.  We can update it later to improve.

File: README.md (hunk 10 above)
Comment (mcvella) [reply README.md:95]: I want them to log in as root and immediately change the passwords... so less steps than going in as non-root, becoming root, etc.

File: README.md (hunk 8 above)
Comment (mcvella) [reply README.md:100]: I think if someone really wants to use this board, securing it is one of the most important things!

File: README.md (hunk 9 above)
Comment (mcvella) [reply README.md:106]: its best practice and take little time...

File: README.md (hunk 11 above)
Comment (mcvella) [reply README.md:113]: I was considering this a practical guide to setting up the board to be successful with Viam.  Leaving unsecured and no wifi and having them dig for how to do these things elsewhere is non-optimal

File: README.md (hunk 12 above)
Comment (mcvella) [reply README.md:116]: Having to look at a router for an IP address that can change each time one is to ssh in is not fun.  This makes it easy for people to ssh in.

File: README.md (hunk 13 above)
Comment (mcvella) [reply README.md:118]: Honestly, they can just run the two commands in the app if they want to just install Viam.  This guide is meant to be a practical setup

File: README.md (hunk 19 above)
Comment (mcvella) [reply README.md:141]: maybe later

File: README.md (hunk 2 above)
Comment (mcvella) [reply README.md]: I found the online docs for this board to be hard to use to find pin numbers, but this tool makes it easy.  I added a comment as to why in the doc
